### DISCUSSION TITLE
crosscompilation on a dedicated built host

### DISCUSSION THREAD

**Discussion by johebll** [spack/spack#43185] [Q&A] [answered]
Hello,

i try to install same packages for the different microarchitectures our cluster partitions have.
I hoped there would be some magic, that would allow to install a package for all the required targets on one dedicated build host, via:
````
spack install zlib@1.3 %gcc@13.2.0 target=zen2
````
but that fails on a broadwell host with:
````
spack.error.SpackError: concretization failed for the following reasons:
   1. Cannot select a single "node_target" for package "zlib"
   2. 'zlib target=zen2' is not compatible with this machine
````
which of course is not too unexpected.

Is there any better option besides repeating the same on a physical "zen2"-node?
I tried to avoid the latter because of the cumbersome workflow...

Best

**Comment by johebll** [comment]
Just found "Cross Compiling" at [https://groups.google.com/g/spack/c/aFsGIhsDKCg/m/OtzR__unCgAJ], and apparently crosscompiling in Spack is not downright impossible. TG mentions that it is practised in 2020.
Since all our microarchitectures are "x86_64", it would not even be the same challange as x86_64/aarch64...

So my hopes are rising someone of you might work like that?

**Comment by dacolombo** [comment, marked as answer]
I think what you are looking for is this option in the concretizer config:

```yaml
concretizer:
  targets:
    host_compatible: false
```

(https://spack.readthedocs.io/en/latest/build_settings.html)

**Comment by johebll** [comment]
Hello @dacolombo,
thank you very much, that is exactly what i was looking for!
And it worked for me.
